test/ccc/core/base/panel/legend/legend-panel-spec.js

```diff
@@ -0,0 +1,184 @@
+          var result = createChart(chartOptions);
+
+          expect(result.legendPanelSize.width).toBe(chartOptions.legendSize.width);
+          expect(result.legendPanelSize.height).toBe(chartOptions.legendSize.height);
+
+          expect(result.legendPanel.isVisible).toBe(true);
+        });
+
+        it("'legendSizeMax' shouldn't set the maximum size of the legend panel if overflow is detected.", function() {
+          var chartOptions = createOptions({legendOverflow: 'collapse', legendSizeMax: size});
+          var result = createChart(chartOptions);
+
+          expect(result.legendPanelSize.width).toBe(0);
+          expect(result.legendPanelSize.height).toBe(0);
+        });
+
+        it("should hide the legend panel when 1 or more legend items don't fit the panel.", function() {
+          var chartOptions = createOptions({legendOverflow: 'collapse', legendSizeMax: size});
+          var result = createChart(chartOptions);
+
+          expect(result.legendPanel.isVisible).toBe(false);
+        });
+
+      }); // Collapse Mode - End
+
+    });
+
+  });
+
+  function createChart(options) {
+    var chart = new pvc.BarChart(options);
+    var dataSpec = datas['relational, series=city|category=date|value=qty, square form'];
+
+    chart.setData.apply(chart, dataSpec);
+
+    chart._create(options);
+    chart.render();
+
+    var legendPanel = chart.legendPanel;
+    return {
+      legendPanel: legendPanel,
+      legendPanelSize: _legendPanelSize(legendPanel._layoutInfo),
+      legendItemCount: _legendItemCount(legendPanel._getRootScene()),
+      legendItemsSizeTotal: _itemsTotalSize(legendPanel)
+    }
+  }
+
+  function createOptions(options) {
+    var chartOptions = {
+      canvas: "pvcBar1",
+      width:  400,
+      height: 400,
+      title:  "Bar chart",
+
+      legend: true,
+      anchor: "top"
+    };
+
+    if('legendOverflow' in options) chartOptions.legendOverflow = options.legendOverflow;
+    if('legendItemCountMax' in options) chartOptions.legendItemCountMax = options.legendItemCountMax;
+    if('legendSize' in options) chartOptions.legendSize = options.legendSize;
+    if('legendSizeMax' in options) chartOptions.legendSizeMax = options.legendSizeMax;
+
+    return chartOptions;
+  }
+
+  function _itemsTotalSize(legendPanel) {
+    var itemsTotalSize = { width: 0, height: 0 };
+    if(!legendPanel.pvLegendPanel) return itemsTotalSize;
+
+    var sections = legendPanel.pvLegendPanel.data().vars.sections;
+    for(var i = 0, L = sections.length; i < L; i++) {
+      itemsTotalSize.width += sections[i].size.width;
+      itemsTotalSize.height += sections[i].size.height;
+    }
+
+    return itemsTotalSize;
+  }
+
+  function _legendPanelSize(layout) {
+    var panelSize = { width: layout.clientSize.width, height: layout.clientSize.height };
+
+    if(panelSize.width > 0) {
+      panelSize.width += layout.realMargins.width + layout.realPaddings.width;
```

Comment:
Please do not use realMargins and realPaddings (they are being removed from another working branch). Use layout.margins and layout.paddings instead. A new version will also already come with layoutInfo.size... but that's not yet usable...